src/Raven.Server/Storage/Schema/Updates/Documents/60000/From50002.cs

```diff
@@ -0,0 +1,132 @@
+using static Raven.Server.Documents.Schemas.Tombstones;
+
+namespace Raven.Server.Storage.Schema.Updates.Documents
+{
+    public class From50002 : ISchemaUpdate
+    {
+        public int From => 50_002;
+
+        public int To => 60_000;
+        
+        public SchemaUpgrader.StorageType StorageType => SchemaUpgrader.StorageType.Documents;
+
+        internal static int NumberOfItemsToMigrateInSingleTransaction = 10_000;
+
+        private long _processedSinceLastCommit;
+
+        public bool Update(UpdateStep step)
+        {
+            InsertIndexValuesFor(step, DocsSchemaBase60, DocsBucketAndEtagSlice,
+                fixedSizeIndex: DocsSchemaBase60.FixedSizeIndexes[AllDocsEtagsSlice],
+                etagPosition: (int)DocumentsTable.Etag);
+
+            InsertIndexValuesFor(step, TombstonesSchemaBase60, TombstonesBucketAndEtagSlice,
+                fixedSizeIndex: TombstonesSchemaBase60.FixedSizeIndexes[AllTombstonesEtagsSlice],
+                etagPosition: (int)TombstoneTable.Etag);
+
+            InsertIndexValuesFor(step, RevisionsSchemaBase60, RevisionsBucketAndEtagSlice,
+                fixedSizeIndex: RevisionsSchemaBase60.FixedSizeIndexes[AllRevisionsEtagsSlice],
+                etagPosition: (int)RevisionsTable.Etag);
+
+            InsertIndexValuesFor(step, TimeSeriesSchemaBase60, TimeSeriesBucketAndEtagSlice,
+                fixedSizeIndex: TimeSeriesSchemaBase60.FixedSizeIndexes[AllTimeSeriesEtagSlice],
+                etagPosition: (int)TimeSeriesTable.Etag);
+
+            InsertIndexValuesFor(step, DeleteRangesSchemaBase60, DeletedRangesBucketAndEtagSlice,
+                fixedSizeIndex: DeleteRangesSchemaBase60.FixedSizeIndexes[AllDeletedRangesEtagSlice],
+                etagPosition: (int)DeletedRangeTable.Etag);
+
+            InsertIndexValuesFor(step, CountersSchemaBase60, CountersBucketAndEtagSlice,
+                fixedSizeIndex: CountersSchemaBase60.FixedSizeIndexes[AllCountersEtagSlice],
+                etagPosition: (int)CountersTable.Etag);
+
+            InsertIndexValuesFor(step, ConflictsSchemaBase60, ConflictsBucketAndEtagSlice,
+                fixedSizeIndex: ConflictsSchemaBase60.FixedSizeIndexes[AllConflictedDocsEtagsSlice],
+                etagPosition: (int)ConflictsTable.Etag);
+
+            InsertIndexValuesFor(step, AttachmentsSchemaBase60, AttachmentsBucketAndEtagSlice,
+                fixedSizeIndex: AttachmentsSchemaBase60.FixedSizeIndexes[AttachmentsEtagSlice],
+                etagPosition: (int)AttachmentsTable.Etag,
+                tableName: AttachmentsMetadataSlice.ToString());
+
+            return true;
+        }
+
+        private void InsertIndexValuesFor(UpdateStep step, TableSchema schema, Slice indexName, TableSchema.FixedSizeKeyIndexDef fixedSizeIndex, int etagPosition, string tableName = null)
+        {
+            var fst = step.ReadTx.FixedTreeFor(fixedSizeIndex.Name, sizeof(long));
+            if (fst.NumberOfEntries == 0)
+                return;
+
```

Comment:
Why do you have this? It is that handle repeated updates?
If so, that isn't valid, that is handled by the schema version marker, not the existence of the tree.

Reply:
this is the index we're using to iterate over the items, e.g. `AllDocsEtags`
if it's empty then there's no items to process

Reply:
What about repeated updates that Oren asked about? Will the update work correctly if the schema update will be aborted midway though and we'll start the server again?

Reply:
Let's remove this code. Doing the `foreach` on empty tree gives us the same behavior, and easier to follow.

Aborted midway should be fine, we'll overwrite existing entries in this manner.

Reply:
tested manually, works fine

Reply:
Cool